server/src/main/java/org/elasticsearch/health/node/DiskHealthIndicatorService.java

```diff
@@ -83,8 +87,18 @@ public String name() {
 
     @Override
     public HealthIndicatorResult calculate(boolean verbose, int maxAffectedResourcesCount, HealthInfo healthInfo) {
+        ClusterState clusterState = clusterService.state();
         Map<String, DiskHealthInfo> diskHealthInfoMap = healthInfo.diskInfoByNode();
         if (diskHealthInfoMap == null || diskHealthInfoMap.isEmpty()) {
+            if (featureService.clusterHasFeature(clusterState, HealthFeatures.SUPPORTS_HEALTH) == false) {
+                return createIndicator(
+                    HealthStatus.GREEN,
+                    "No disk usage available. This probably means a cluster upgrade is in progress.",
```

Comment:
We should be a little more confident in our message here, what do you think of:
```suggestion
                    "No disk usage available. The cluster currently has mixed versions (an upgrade may be in progress.)",
```